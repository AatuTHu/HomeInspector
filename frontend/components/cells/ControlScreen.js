import { View, Text, ScrollView} from "react-native"
import { styles } from "../../styles/styles"
import  variables  from "../../env"
import { useState } from "react"
import Buttons from "../atoms/Buttons"
import SaveSensorLocation from "../molecules/SaveSensorLocation"
import DropDown  from "../molecules/DropDown"
import StartMeasuring from "../molecules/StartMeasuring"
import LightSwitch from "../molecules/LightSwitch"

export default function ControlScreen({temperatureLocation,humidityLocation,setScreen}) {

  const { humidityLocationURL,humidityStartURL,temperatureStartURL,temperatureLocationURL} = variables

  const options = [
    {id : 1 ,name : "humidity", value : "humidity"},
    {id : 2, name : "temperature", value : "temperature"}
  ]

  let location
  let startURL
  let locationURL

  const [selected, setSelected] = useState("")
  
  const showAll = () => { 
    setScreen(4);
  }

  if(selected === 'temperature') {
    location = temperatureLocation
    startURL = temperatureStartURL
    locationURL = temperatureLocationURL 
  } else if (selected === 'humidity') {
    location = humidityLocation
    startURL = humidityStartURL
    locationURL = humidityLocationURL
  }

  return (
    <ScrollView>
      <View style = { styles.controlContainer }>
          <DropDown selected={selected} setSelected={setSelected} location = { location } options={options}/>
          { selected ? (<>
            <SaveSensorLocation location={location} locationURL = { locationURL }/>
              <StartMeasuring startURL={startURL}/>
          </>) : (<></>)}
       
          <LightSwitch/>
          <Buttons text = "See all the measurements" type = "update" onPress={ showAll }/>
      </View>
    </ScrollView>
  )
}