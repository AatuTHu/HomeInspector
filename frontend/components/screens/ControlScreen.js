import { View, Text, ScrollView} from "react-native"
import  variables  from "../../env"
import { styles } from "../../styles/styles"
import { useState } from "react"
import Buttons from "./atoms/Buttons"
import InfoBox from "./atoms/InfoBox"
import InputBox from "./atoms/InputBox"


export default function Control({ssid, location, setScreen}) {

  const { apiKey,locationURL,lightURL,startURL } = variables

  const [newLocation, setNewLocation] = useState("");
  const [statusText, setStatusText] = useState("")
  const [timer, setTimer] = useState(0)
  const [start, setStart] = useState(false)
  const [lights, setLights] = useState(false)

  const lightSwitch = async() => {
    setLights(!lights)
    let mode = lights ?  0 : 1
    await fetch(lightURL, {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: `apiKey=${apiKey}&lightMode=${mode}`,
    }).then( res => {
      if(res.status === 200) {
        lights ? setStatusText("Lights Off") : setStatusText("Lights On")
      } else {
        setStatusText("There was an error, Try again later")
        setLights(false)
      }
    }).catch( error => {
      console.log(error)
    })
  }

  const saveLocation = async() => {
   await fetch(locationURL, {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: `apiKey=${apiKey}&location=${newLocation}`,
    }).then( res => {
      if(res.status === 200) {
        setStatusText("Location switched")
      } else {
        setStatusText("There was an error, Try again later")
      }
    }).catch( error => {
      console.log(error)
    })
  }

  const startMeasuring = async() => {
    setStart(!start)
    let mode = start? 0 : 1
    await fetch(startURL, {
      method: "POST",
      headers: { "Content-Type": "application/x-www-form-urlencoded" },
      body: `apiKey=${apiKey}&measuringMode=${mode}`,
    }).then( res => {
      if(res.status === 200) {
        start ? setStatusText("Measurements stopped") : setStatusText("Measurements started")
      } else {
        setStatusText("There was an error, Try again later")
      }
    }).catch( error => {
      console.log(error)
    })
  }

  const showAll = () => { 
    setScreen(4);
  }

  const deleteAll = () => {
    
  }

  return (
    <>
    <ScrollView>
      <View style = { styles.controlContainer }>
        <InfoBox text = "Wi-Fi ssid" textStyle = 'headingText' type = 'heading'/>
          <InfoBox text = { ssid } type =  'small' textStyle = 'subHeadingText' />
            <InfoBox text = "Device location" textStyle = 'headingText' type = 'heading'/>  
          <InputBox placeholder= "where ..." text = { location } newText = { setNewLocation } onSubmitEditing={ saveLocation }/>
        <Buttons text = "save" type = "save" onPress={ saveLocation } />

      <View style = { styles.block}>
        <Text style = { styles.statusText }>{statusText}</Text>
      </View>

      <View style = { styles.buttonRowContainer }>
        <Buttons text = "start" type = "start" onPress={ startMeasuring }/>
          <InfoBox text = { timer } type = 'status' textStyle = 'subHeadingText' />
      </View>

      <View style = { styles.buttonRowContainer }>
        <Buttons text = "Lights on/off" type = "start" onPress={ lightSwitch }/>
        <View style = { styles.statusBox }>
            <Text style = { styles.subHeadingText }>{lights ? (<Text style = {{"color" : "yellow"}} >On</Text>) : (<Text>Off</Text>)}</Text>
        </View>
      </View>

          <Buttons text = "See all the measurements" type = "update" onPress={ showAll }/>
        <Buttons text = "Delete measurements" type = "delete" onPress={ deleteAll }/>
      </View>
    </ScrollView>
  </>
  )
}