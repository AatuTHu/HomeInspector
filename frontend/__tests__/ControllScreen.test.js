import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { enableFetchMocks } from 'jest-fetch-mock'
import ControlScreen from '../components/cells/ControlScreen';

enableFetchMocks();

describe('ControlScreen', () => {
  
  it('renders correctly', () => {
    const screen = render(<ControlScreen/>);
    expect(screen).toMatchSnapshot();
  });

});