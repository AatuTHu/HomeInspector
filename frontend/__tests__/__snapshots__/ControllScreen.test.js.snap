// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlScreen renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "borderRadius": 22,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 20,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 20,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Sensor
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Select a Sensor..
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "black",
                "fontSize": 20,
                "opacity": 0.5,
                "padding": 10,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "marginTop": 1,
            "width": "95%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "lightgrey",
              "borderRadius": 20,
              "borderWidth": 1,
              "opacity": 1,
              "padding": 5,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            lights
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "lightgrey",
              "borderRadius": 22,
              "justifyContent": "center",
              "width": "30%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            <Text>
              Off
            </Text>
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "lightblue",
            "borderRadius": 20,
            "borderWidth": 1,
            "marginTop": 20,
            "opacity": 1,
            "padding": 10,
            "width": "95%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          See all the measurements
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
